### ISSUE TITLE
Add support for customizing ZookeeperCluster properties

### ISSUE THREAD

**Issue by ziodave** [apache/solr-operator#148]
**Is your feature request related to a problem? Please describe.**

I am uploading several files related to a configset, these files are written to Zookeeper and some of them are over 1 MB. By default Zookeeper has a limit of 1 MB, I need to customize Zookeeper properties to increase the limit.

**Describe the solution you'd like**

Add a spec with Zookeeper options similar to `solrOpts`, e.g. `zookeeperOpts` or `zkOpts`. Among those `zkOpts.env` (or `zkOpts.pod.env`) to customize the the Zookeeper environment variables.

**Describe alternatives you've considered**

ATM I am creating a `ZookeeperCluster`, setting `spec.pod.env` and referencing that cluster from `SolrCloud`.

**Additional context**

For reference see here:
* [Chat on Slack](https://kubernetes.slack.com/archives/CQSNS615F/p1603210359006400)
* [Solr Cloud can't upload large config files ( > 1MB) to Zookeeper](https://issues.apache.org/jira/browse/SOLR-4793)
* [Configuring jute.maxbuffer on ZooKeeper Nodes](https://lucene.apache.org/solr/guide/8_5/setting-up-an-external-zookeeper-ensemble.html#configuring-jute-maxbuffer-on-zookeeper-nodes)
* [Experimental Options](https://zookeeper.apache.org/doc/r3.4.5/zookeeperAdmin.html#Experimental+Options%2FFeatures)

**Comment by ziodave** [comment]
In my specific case as I am altering the `jute.maxbuffer` property it is important to highlight that the same setting must be applied to Solr:

>  Note that this configuration, which is required both for ZooKeeper server(s) and **for all clients** that connect to the server(s), must be the same everywhere it is specified.

https://lucene.apache.org/solr/guide/8_5/setting-up-an-external-zookeeper-ensemble.html#increasing-the-file-size-limit

As a reference this is my yaml:

```
apiVersion: "zookeeper.pravega.io/v1beta1"
kind: "ZookeeperCluster"
metadata:
  name: solr-solrcloud-zookeeper
spec:
  replicas: 3
  pod:
    env:
      # The `jute.maxbuffer` property must match the one set in solrOpts.
      - name: JVMFLAGS
        value: "-Djute.maxbuffer=52428800"
---
apiVersion: solr.bloomberg.com/v1beta1
kind: SolrCloud
metadata:
  name: solr
spec:
  replicas: 3
  solrImage:
    tag: "8.5.2"
  solrJavaMem: "-Xms1G -Xmx1G"
  # Add -Djute.maxbuffer to increase the size that ZooKeeper can store
  # https://issues.apache.org/jira/browse/SOLR-4793
  # https://zookeeper.apache.org/doc/r3.4.5/zookeeperAdmin.html#sc_configuration
  solrOpts: -Dblob.max.size.mb=50 -Djute.maxbuffer=52428800
  dataPvcSpec:
    storageClassName: azurefile
    resources:
      requests:
        storage: "5Gi"
  backupRestoreVolume:
    persistentVolumeClaim:
      claimName: "solr-backup-pvc"
  zookeeperRef:
    connectionInfo:
      internalConnectionString: solr-solrcloud-zookeeper-0.solr-solrcloud-zookeeper-headless.solr:2181,solr-solrcloud-zookeeper-1.solr-solrcloud-zookeeper-headless.solr:2181,solr-solrcloud-zookeeper-2.solr-solrcloud-zookeeper-headless.solr:2181
```

**Comment by dalbani** [comment]
I'm also looking for ways to configure the ZookeeperCluster that it is created by the Solr operator.
For example the ability to set environment variables, in order to configure logging: https://github.com/pravega/zookeeper-operator/issues/252.
But also the ability to set my own labels and annotations on the ZooKeeper pods, as supported by the ZookeeperCluster CRD: https://github.com/pravega/zookeeper-operator/blob/master/deploy/crds/zookeeper.pravega.io_zookeeperclusters_crd.yaml#L2055.

**Comment by HoustonPutman** [comment]
Forgot to update this, but this has been implemented since `v0.3.0`, released in May 2021.
